src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+
+/**
+ * Represents a parser for user input given to Duke chatbot.
+ *
+ * @author Conrad
+ */
+public class Parser {
+
+    /**
+     * Parses the task number from the user input command into its <code>TaskList</code> index.
+     *
+     * @param userInput String input provided by user.
+     * @return An integer representing the index of the task in the <code>TaskList</code>.
+     * @throws DukeException If input is not of the type [command] [taskNumber]
+     */
+    public static int parseTaskNumber(String userInput) throws DukeException {
+        String[] parsedInput = userInput.split(" ");
+        String taskNumberString = parsedInput[1];
+        try {
+            return Integer.parseInt(taskNumberString) - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("   " + taskNumberString + " is not a valid task number.\n");
+        }
+
+    }
+
+    /**
+     * Parses the user input into a <code>Todo</code> task.
+     *
+     * @param userInput String input provided by user.
+     * @return An <code>Todo</code> task with the specified description.
+     */
+    public static Todo parseTodoTask(String userInput) {
+        String[] parsedUserResponse = userInput.split(" ");
+        String newTaskDescription = String.join(" ",
+                Arrays.copyOfRange(parsedUserResponse, 1, parsedUserResponse.length));
+        return new Todo(newTaskDescription);
+    }
+
+    /**
+     * Parses the user input into an <code>Event</code> task.
+     *
+     * @param userInput String input provided by user.
+     * @return An <code>Event</code> task with the specified description and time of occurrence.
+     */
+    public static Event parseEventTask(String userInput) {
+
+        String[] parsedUserResponse = userInput.split(" ");
+        int atSeparationIndex = Arrays.asList(parsedUserResponse).indexOf("/at");
+        String newTaskDescription = String.join(" ",
+                Arrays.copyOfRange(parsedUserResponse,
+                        1,
+                        atSeparationIndex));
+        String newEventTime = String.join(" ",
+                Arrays.copyOfRange(parsedUserResponse,
+                        atSeparationIndex + 1,
+                        parsedUserResponse.length));
+        return new Event(newTaskDescription, newEventTime);
+    }
+
```

Comment:
Using the same idea as the comment on `parseTodoTask`, and assuming we are trying to get the description (which is before /at) and the time (which is after /at) maybe we could do this? If you agree it is easier to understand.
```java
String[] parsedUserResponse = userInput.replace("/event").split("/at");
String newTaskDescription = parsedUserResponse[0];
String newEventTime = parsedUserResponse[1];
```